avro-schema-processor/src/main/java/io/micronaut/avro/visitor/AvroSchemaVisitor.java

```diff
@@ -0,0 +1,433 @@
+                 "java.time.OffsetTime" -> {
+                avroSchema.setType(Type.LONG);
+                if (context.useTimestampMillisLogicalType()) {
+                    avroSchema.setLogicalType(LogicalType.TIMESTAMP_MILLIS);
+                }
+            }
+            case "java.time.Duration" -> {
+                if (context.useLogicalTypes()) {
+                    avroSchema.setType(Type.FIXED);
+                    avroSchema.setSize(12); // Duration is 12 bytes
+                    avroSchema.setLogicalType(LogicalType.DURATION);
+                } else {
+                    avroSchema.setType(Type.LONG);
+                }
+            }
+            default -> avroSchema.setType(Type.STRING);
+        }
+    }
+
+    /**
+     * Sets numeric type for the schema based on the Java type.
+     *
+     * @param type       The class element representing the type
+     * @param avroSchema The schema to configure
+     * @param context    The Avro schema creation context
+     */
+    private static void setNumericType(ClassElement type, AvroSchema avroSchema, AvroSchemaContext context) {
+        switch (type.getName()) {
+            case "java.lang.Integer" -> avroSchema.setType(Type.INT);
+            case "java.lang.Float" -> avroSchema.setType(Type.FLOAT);
+            case "java.lang.Double" -> avroSchema.setType(Type.DOUBLE);
+            case "java.lang.Long" -> avroSchema.setType(Type.LONG);
+            case "java.math.BigDecimal" -> {
+                if (context.useDecimalLogicalType()) {
+                    avroSchema.setType(Type.BYTES);
+                    avroSchema.setJavaClass(type.getCanonicalName());
+                    avroSchema.setLogicalType(LogicalType.DECIMAL);
+                } else {
+                    avroSchema.setType(Type.STRING);
+                }
+
+            }
+            case "java.math.BigInteger" -> {
+                // BigInteger doesn't have a specific logical type
+                avroSchema.setType(Type.STRING);
+                avroSchema.setUnsupported(true);
+                avroSchema.setJavaClass(type.getCanonicalName());
+            }
+            case "java.lang.Short" -> {
+                avroSchema.setUnsupported(true);
+                avroSchema.setType(Type.INT);
+                avroSchema.setJavaClass("java.lang.Short");
+            }
+            case "java.lang.Byte" -> {
+                avroSchema.setUnsupported(true);
+                avroSchema.setType(Type.INT);
+                avroSchema.setJavaClass("java.lang.Byte");
+            }
+
+            default -> avroSchema.setType(Type.DOUBLE);
+        }
+    }
+
+    private static void setBeanSchemaFields(ClassElement element, VisitorContext visitorContext, AvroSchemaContext context, AvroSchema avroSchema) {
+        avroSchema.setType(Type.RECORD);
+
+        avroSchema.setNamespace(element.getPackageName());
```

Comment:
Perhaps it should check that it has either `Serdeable` or `Avro` annotation?

The Avro annotation probably could map to Serdeable, as well.

Reply:
Additionally, it might be a good idea to have some check for circular type usage, e.g:
```
record Person(List<Person> children);
```